CHANGELOG.md

```diff
@@ -1,4 +1,12 @@
-# [Unreleased](https://github.com/pybamm-team/PyBaMM)
+# [v0.4.0](https://github.com/pybamm-team/PyBaMM/tree/v0.4.0) - 2021-03-27
+
+This release introduces:
+- several new models, including reversible and irreversible plating submodels, submodels for loss of active material, Yang et al.'s (2017) coupled SEI/plating/pore clogging model, and the Newman-Tobias model
+- internal optimizations for solving models, particularly for simulating experiments, with more accurate event detection and more efficient numerical methods and post-processing
+- parallel solutions of a model with different inputs
+- a cleaner installation process for Mac, no longer requiring a Homebrew installation of Sundials
```

Comment:
A minor point: maybe mention that this is true when installing from PyPI, otherwise maybe people installing from the repo will be confused. (Sundials is still required when installing from the repo)
> a cleaner installation process for Mac when installing from PyPI, no longer requiring a local installation of Sundials